packages/material-ui/src/MuiComponentContext/useMuiComponent.ts

```diff
@@ -0,0 +1,16 @@
+import { useContext } from 'react';
+import MaterialUIContextProps from '../Theme/MaterialUIContextProps';
+import Mui5ContextProps from '../Theme5/Mui5ContextProps';
+import MuiComponentContext from './MuiComponentContext';
+
+export function useMuiComponent(): MaterialUIContextProps | Mui5ContextProps {
+  const muiComponents = useContext(MuiComponentContext);
+
+  if (!muiComponents) {
+    throw new Error(
```

Comment:
This provides good feedback to the consumers if they don't have one of the MUI version peer deps installed.